### PR TITLE
ADD: save model option

### PR DESCRIPTION
**Pull request by luizduducoelho**
Add argument to save model with the best validation metric

### CODE REVIEW

File: miyagi_trainer/train.py [hunk 1]
```diff
@@ -146,6 +148,21 @@ def train_model(model,
...
+                    model_folder = wandb.run.name if track_experiment else \
+                                   datetime.datetime.now().strftime('%Y-%m-%d_%H-%M-%S')
+                    if not os.path.exists(model_folder):
+                        os.mkdir(model_folder)
+                    torch.save({
+                            'epoch': epoch,
+                            'model_state_dict': model.state_dict(),
+                            'optimizer_state_dict': optimizer.state_dict(),
+                            'loss': loss,
+                            }, os.path.join(model_folder, "best_model_save.pth"))
```
Comment (gustavofuhr) [inline miyagi_trainer/train.py:164]: I will call by default `best_saved_model.pth` and whats the best extension here `.pth` or `pt`?

File: miyagi_trainer/train.py [hunk 2]
```diff
@@ -260,6 +278,9 @@ def train(args):
     parser.add_argument("--optimizer", default="sgd") # possible adam, adamp and sgd
     parser.add_argument("--weight_decay", type=float, default=1e-4)
 
+    # options for model saving
+    parser.add_argument("--save_model", action=argparse.BooleanOptionalAction)
```
Comment (gustavofuhr) [inline miyagi_trainer/train.py:282]: I would change the param to `--save_best_model`.

**Review by gustavofuhr — CHANGES_REQUESTED** [review]

File: miyagi_trainer/train.py (hunk 1 above)
Comment (luizduducoelho) [reply miyagi_trainer/train.py:164]: It appears that `.pt` is preferable, according to https://discuss.pytorch.org/t/what-does-pth-tar-extension-mean/36697/3. 
I will modify it.
